File: e2e/elixir/test/strategy_test.exs
# E2e tests for category: strategy
defmodule E2e.StrategyTest do
  use ExUnit.Case, async: true

  describe "strategy_best_first_seed" do
    test "BestFirst strategy always processes the seed URL first" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.crawl.pages_crawled) == 3
      assert String.contains?(result.strategy.first_page_url_contains, "/")
    end
  end

  describe "strategy_bfs_default_order" do
    test "BFS strategy visits pages in breadth-first order" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.crawl.pages_crawled) == 5
      assert String.trim(result.strategy.crawl_order) == ["/", "/a", "/b", "/a/1", "/b/1"]
    end
  end

  describe "strategy_dfs_depth_first" do
    test "DFS strategy visits pages in depth-first order" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.crawl.pages_crawled) == 5
      assert String.trim(result.strategy.crawl_order) == ["/", "/b", "/b/1", "/a", "/a/1"]
    end
  end
end

File: e2e/java/src/test/java/dev/kreuzberg/e2e/StrategyTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import dev.kreuzberg.kreuzcrawl.Kreuzcrawl;

/** E2e tests for category: strategy. */
class StrategyTest {
    @Test
    void testStrategyBestFirstSeed() throws Exception {
        // BestFirst strategy always processes the seed URL first
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'crawl.pages_crawled' not available on result type
        // skipped: field 'strategy.first_page_url_contains' not available on result type
    }

    @Test
    void testStrategyBfsDefaultOrder() throws Exception {
        // BFS strategy visits pages in breadth-first order
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'crawl.pages_crawled' not available on result type
        // skipped: field 'strategy.crawl_order' not available on result type
    }

    @Test
    void testStrategyDfsDepthFirst() throws Exception {
        // DFS strategy visits pages in depth-first order
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        // skipped: field 'crawl.pages_crawled' not available on result type
        // skipped: field 'strategy.crawl_order' not available on result type
    }

}
